docs(common): Add Authentication with NuGet Keys (#1071)
telerik/blazor-docs: deployment/nuget-keys.md

---
title: Restoring NuGet Packages in CI
page_title: Restoring NuGet Packages in CI

> description: "Get started with Telerik UI for Blazor and use NuGet Keys to authenticate with the Telerik NuGet server and restore packages in your CI or desktop environment."

Reviewer: Do we need quotation marks here? I haven't seen them being used elsewhere.

Author: You're right, the Blazor docs don't use quotation marks in the description.
```suggestion
description: Get started with Telerik UI for Blazor and use NuGet Keys to authenticate with the Telerik NuGet server and restore packages in your CI or desktop environment.
```

slug: installation/nuget-keys
position: 7
---

# Restoring NuGet Packages in Your CI Workflow

> This article provides an overview of the most popular approaches for using token-based authentication to restore Telerik NuGet packages in your CI (Continuous Integration) workflow.

Reviewer: ```suggestion
This article provides an overview of the most popular ways to use token-based authentication to restore Telerik NuGet packages in your Continuous Integration (CI) workflow.
```

Author: For completeness, the article mentions one other way—basic authentication. But it handles in detail only the token-based authentication through NuGet keys and the workflows where these keys can be used.

```suggestion
This article provides an overview of the most popular ways for using token-based authentication to restore Telerik NuGet packages in your Continuous Integration (CI) workflow.
```

The Telerik NuGet server allows you to authenticate by using two methods:

* Basic authentication by providing your Telerik user name and password.
* Token-based authentication by providing a NuGet Key.

When you need to restore Telerik NuGet packages as part of your CI, using NuGet keys is the more secure way to authenticate. This method does not require you to provide your Telerik username and password anywhere in the CI workflow.

Unlike your Telerik credentials, a NuGet Key has a limited scope and can be used only with the Telerik NuGet server. If any of your NuGet keys is compromised, you can quickly delete it and create a new one.

## Generating NuGet Keys

1. Go to the [**Manage NuGet Keys**](https://www.telerik.com/account/downloads/nuget-keys) page in your Telerik account.

    ![Manage NuGet Keys](../deployment/images/manage-nuget-keys.png)

> 1. To create a new key, select the **Generate New Key** button.

Reviewer: ```suggestion
1. To create a new key, click on **Generate New Key**.
```

1. Enter a name for the NuGet Key, and then select **Generate Key**.

1. To copy the key, select **Copy and Close**. Once you close the window, you can no longer copy the generated key. For security reasons, the **NuGet Keys** page displays only a portion of the key.

    ![Copy Generated NuGet Key](../deployment/images/copy-nuget-key.png)

## Storing a NuGet Key

> Never check in a NuGet Key with your source code or leave it publicly visible in plain text, for example, as a raw key value in a `nuget.config` file. A NuGet Key is valuable as bad actors can use it to access the NuGet packages that are licensed under your account. A potential key abuse could lead to a review of the affected account.

To protect the NuGet Key, store it as a secret environment variable. The exact steps depend on your workflow:

> 

Reviewer: Maybe we should use a table here, to highlight the different environments. Or at least bold them.

Author: I am hesitant about using a table with three rows. There is no need to bold them for emphasis as these workflows are at the beginning of the list items.
